### PR TITLE
Don't throw 500s when auth has gone wrong

### PR DESCRIPTION
**Pull request by eleanorreem**
- I noticed that when you were unauthorised and tried to create an access code, the api returned with a 500.
- I decided to fix this issue with the auth first and return with a 403 - forbidden. 
- I removed errors thrown because this makes it look like it is an error on our behalf when it really isn't. 500s should be for cases that are really bad, not lack of authorisation. 
- I added a test for the auth guard
- I added a test for generate access code
- Note that I have decided to ignore the type issue for now because I spent too much time on it already.

### CODE REVIEW

File: src/partner-admin/partner-admin-auth.guard.ts [hunk 1]
```diff
@@ -8,26 +8,31 @@ export class PartnerAdminAuthGuard implements CanActivate {
   constructor(private authService: AuthService, private usersRepository: UserRepository) {}
 
   async canActivate(context: ExecutionContext): Promise<boolean> {
-    const request = context.switchToHttp().getRequest<Request>();
-
+    const request = await context.switchToHttp().getRequest<Request>();
```
Comment (swetha-charles) [inline src/partner-admin/partner-admin-auth.guard.ts:11]: How come this now needs an await?

File: src/partner-admin/partner-admin-auth.guard.ts [hunk 2]
```diff
@@ -8,26 +8,31 @@ export class PartnerAdminAuthGuard implements CanActivate {
...
     const { authorization } = request.headers;
-
     if (!authorization) {
-      throw new UnauthorizedException('Unauthorized: missing required Authorization token');
+      return false;
```
Comment (swetha-charles) [inline src/partner-admin/partner-admin-auth.guard.ts:14]: Good idea 👍

**Review by swetha-charles — APPROVED** [review]
Looks good. 

Could you explain why the await is needed? Apart from that, lgtm

File: src/partner-admin/partner-admin-auth.guard.ts (hunk 1 above)
Comment (eleanorreem) [reply src/partner-admin/partner-admin-auth.guard.ts:11]: Tested it and this was a mistake on my part haha. Good catch. I made the mock wrong. 👀  Lesson in that you've got to get those mocks right 😬

Comment (swetha-charles) [reply src/partner-admin/partner-admin-auth.guard.ts]: Ahaha fair - mocking is quite tricky tbh!

**Review by swetha-charles — APPROVED** [review]
Lgtm 👍

**Pull request merged**